### PR TITLE
Sso

### PR DESCRIPTION
**Pull request by CuriouzK0d3r**
### Description of PR

Added Google SSO

What component(s) does this PR affect?

  - [ ] Back-End (Node.js, Next.js)
  - [x] Front-End (React.js)
  - [ ] Theming (CSS, React theme)
  - [ ] Build System
  - [ ] Docs
  
  
  
  Does the PR require changes on other components? If yes, please mark the components:
  
  - [ ] Back-End (Node.js, Next.js)
  - [ ] Front-End (React.js)
  - [ ] Theming (CSS, React theme)
  - [ ] Build System
  - [ ] Docs
  
  
  
### Related Issue
  
  
  
  
  
  
  Resolves #
  
### Type
  
  - [ ] Bug fix (non-breaking change which fixes an issue)
  - [ ] New feature (non-breaking change which adds functionality)
  - [ ] Breaking change (fix or feature that would cause existing functionality to change)
  - [ ] Docs update
  - [ ] None of the above
  
  
  
### Checklist:
  
  - [ ] I have read the **[contributing guide](https://github.com/FORTH-ICS-INSPIRE/artemis/blob/master/CONTRIBUTING.md)** and my code conforms to the guidelines.
  - [ ] This change requires a change in the documentation.
  - [ ] I have updated the documentation accordingly.
  >)`>>>)>

### CODE REVIEW

**Review by vkotronis — COMMENTED** [review]
LGTM at first sight, could you also write some documentation on using the new feature (+ screenshots if you have)? Also how can the admin limit the users that enter the system, are the google-logged first-time users entering the "pending approval" workflow as discussed? Otherwise new google logged-in users can become users without any admin approval.

**Comment by CuriouzK0d3r** [comment]
> LGTM at first sight, could you also write some documentation on using the new feature (+ screenshots if you have)? Also how can the admin limit the users that enter the system, are the google-logged first-time users entering the "pending approval" workflow as discussed? Otherwise new google logged-in users can become users without any admin approval.

A new user using the Google SSO will be added with 'pending' privileges. So it is like creating a new user with credentials.

**Comment by vkotronis** [comment]
> Screenshot from 2022-04-11 17-00-46

Maybe instead of "Login with Google" we should have a "Login via SSO" button which, when pressed, shows the available SSO login options in drop-down? We can start with a single one (Google), it seems more scalable to treat it this way I think (plus we may add more SSO options in the drop-down menu in the future)

**Comment by CuriouzK0d3r** [comment]
Screenshot from 2022-04-11 20-26-38
Screenshot from 2022-04-11 20-26-25

What do you think?

File: .env [hunk 1]
```diff
@@ -28,6 +28,10 @@ LDAP_GROUP_SEARCH_ATTRIBUTES=mail,uid
 LDAP_EMAIL_FIELDNAME=mail
 LDAP_ADMIN_GROUP=admin_staff
 
+NEXT_PUBLIC_GOOGLE_ENABLED=false
```
Comment (vkotronis) [inline .env:31]: could you also open a small PR in the backend too to place the env vars? Both in `.env` and `docker-compose` mapping. We can merge that one first and then update the frontend (general approach when we want to add new feature and env vars)

**Comment by vkotronis** [comment]
> Screenshot from 2022-04-11 20-26-38 Screenshot from 2022-04-11 20-26-25
> 
> What do you think?

Yes that is fine

File: apps/artemis-web/components/login/login.tsx [hunk 2]
```diff
@@ -215,6 +221,41 @@ const Login = (props: any): any => {
                 Login with LDAP
               </Button>
             )}
+            {(hasGoogle === 'true' &&
```
Comment (vkotronis) [inline apps/artemis-web/components/login/login.tsx:224]: why `hasGoogle` and not `hasSSO`?

**Comment by vkotronis** [comment]
As discussed, we need also documentation on this:
1. Build Google app and get env vars
2. New user login
3. Existing user login
+ screenshots
Please add in-PR

**Comment by vkotronis** [comment]
@CuriouzK0d3r could you check if the bug with the mongo-stored user is resolved and update the PR with the `master` base? Then it can be merged I think, LGTM

**Review by vkotronis — APPROVED** [review]
LGTM

**Pull request merged**